src/thing/Portal.cpp

```diff
@@ -50,13 +57,44 @@ void Portal::Start() {
           }
         }
       }
+      function scan() {
+        document.getElementById('scanning').style.display='inline';
+        var xhr = new XMLHttpRequest();
+        xhr.open("GET", "/wifi/scan", true);
+        xhr.onreadystatechange = function () {
+          if (xhr.readyState==4 && xhr.status==200) {
+           load_networks(JSON.parse(xhr.responseText));
+           document.getElementById('scanning').style.display='none';
+          }
+        }
+        xhr.send();
+      }
+      function load_networks(networks) {
+        select = document.getElementById('networks');
+        select.innerHtml = '';
+        networks.sort(function(a, b) {
+          return b.rssi - a.rssi;
+        });
+        networks.forEach(function(network) {
+          option = document.createElement('option');
+          option.value = network.ssid;
+          option.text = network.ssid + ' (' + network.rssi +  ')';
+          select.add(option);
+        });
+        select.style.display='inline';
+      }
+      function set_network(select) {
+        document.getElementById('ssid').value = select[select.selectedIndex].value;
+      }
       </script>
       </head>
       <body>
       <div>Host: <input id='host'></div>
       <div>Auth: <input id='auth'></div>
       <div>Path: <input id='path'></div>
-      <div>Wifi SSID: <input id='ssid'></div>
+      <div>Wifi SSID: <input id='ssid'><button onclick='scan();'>scan</button></div>
```

Comment:
did you consider attaching the onclick from the js and keeping the <dom> js free?